src/aktualizr_primary/secondary_config.cc

```diff
@@ -0,0 +1,91 @@
+#include <fstream>
+#include <iostream>
+#include <unordered_map>
+
+#include "logging/logging.h"
+#include "secondary_config.h"
+
+namespace Primary {
+
+const char* const IPSecondaryConfig::Type = "ip";
+
+SecondaryConfigParser::Configs SecondaryConfigParser::parse_config_file(const boost::filesystem::path& config_file) {
+  if (!boost::filesystem::exists(config_file)) {
+    throw std::invalid_argument("Specified config file doesn't exist: " + config_file.string());
+  }
+
+  auto cfg_file_ext = boost::filesystem::extension(config_file);
+  std::shared_ptr<SecondaryConfigParser> cfg_parser;
+
+  if (cfg_file_ext == ".json") {
+    cfg_parser = std::make_shared<JsonConfigParser>(config_file);
+  } else {  // add your format of configuration file + implement SecondaryConfigParser specialization
+    throw std::invalid_argument("Unsupported type of config format: " + cfg_file_ext);
+  }
+
+  return cfg_parser->parse();
+}
+
+/*
+config file example
+
+{
+  "ip": [
+          {"addr": "127.0.0.1:9031"},
+          {"addr": "127.0.0.1:9032"}
+  ],
+  "socketcan": [
+          {"key": "value", "key1": "value1"},
+          {"key": "value", "key1": "value1"}
+  ]
+}
+
+*/
+
+JsonConfigParser::JsonConfigParser(const boost::filesystem::path& config_file) {
+  assert(boost::filesystem::exists(config_file));
+  std::ifstream json_file_stream(config_file.string());
+  Json::Reader json_reader;
+
+  if (!json_reader.parse(json_file_stream, root_, false)) {
+    throw std::invalid_argument("Failed to parse secondary config file: " + config_file.string() + ": " +
+                                json_reader.getFormattedErrorMessages());
+  }
+}
+
+SecondaryConfigParser::Configs JsonConfigParser::parse() {
+  Configs res_sec_cfg;
+
+  for (Json::ValueIterator it = root_.begin(); it != root_.end(); ++it) {
+    std::string secondary_type = it.key().asString();
+
+    if (sec_cfg_factory_registry_.find(secondary_type) == sec_cfg_factory_registry_.end()) {
+      LOG_ERROR << "Unsupported type of sescondary config was found: `" << secondary_type
+                << "`. Ingoring it and continue with parsing of other secondary configs";
```

Comment:
Typo: "Ignoring". And actually "continuing" would be better than "continue".

Reply:
Sure.